eric/ starter+specific-date

Adventurer-E opened the pull request:
### Description
<!--- Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change. --->

### Reason for Change
<!--- Why is this change being made? By default, just provide the Linear ticket ID. You may add extra context if you made changes that were not specified in the ticket. --->

<!--- Please delete options that are not relevant for linking your ticket. You likely only need one of these for your PR --->
Fixes CFD-

Completes CFD-165,166,151

Contributes to CFD-

### Type of change
<!--- Please delete options that are not relevant. --->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation update
- [ ] Other

### Testing
<!--- Please describe the tests that you ran to verify your changes. This may be a screenshot of the change that you made. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration --->

- [X] Test A: Functionalities work on Expo.
- [ ] Test B

sarinali reviewed (changes requested):
Few things: thanks for your work! Can you please include a screenshot of any testing you did? Also you have some conflicts, make sure you address those

Adventurer-E commented:
My information is shown successfully.
<img width="519" alt="Screenshot 2024-11-03 at 11 46 49 AM" src="https://github.com/user-attachments/assets/8af673ee-9f73-4d0f-921e-d23575cf8f01">

Also, I clicked upvote for me and for Carlos, and as shown above, the numbers are returned correctly. Prisma also records the number of clicks I've done:
<img width="1392" alt="Screenshot 2024-11-03 at 11 47 38 AM" src="https://github.com/user-attachments/assets/bb4bff9f-244e-4d15-adbe-10404e311f1c">

Adventurer-E commented:
For now, the getSchedule and getDailySchedule works on API panel.
I manually added a record for 2024-11-03.
<img width="1624" alt="Screenshot 2024-11-03 at 1 32 34 PM" src="https://github.com/user-attachments/assets/d2efbefc-a869-49db-a725-194f4df744c7">
Tests:
getSchedule on 2024-11-01 (should return the record on 11-03 because getSchedule is supposed to find all records in the week 11-01 to 11-08)
<img width="1624" alt="Screenshot 2024-11-03 at 1 32 43 PM" src="https://github.com/user-attachments/assets/cd6fe82b-ea84-45d8-94b9-d7ed3927c1a0">
getDailySchedule on 2024-11-03
<img width="1624" alt="Screenshot 2024-11-03 at 1 32 51 PM" src="https://github.com/user-attachments/assets/6cf7c6c3-e350-4b90-bed2-90431a732390">
getDailySchedule on 2024-11-02 (should return empty)
<img width="1624" alt="Screenshot 2024-11-03 at 2 18 10 PM" src="https://github.com/user-attachments/assets/1cc9e497-33de-40a0-9060-62c157b9b788">
getDailySchedule on an invalid date string
<img width="1624" alt="Screenshot 2024-11-03 at 1 33 00 PM" src="https://github.com/user-attachments/assets/73b190a2-2ebb-4773-b162-7cb9d6c20afc">

Some notes:
1. The input format used to be z.date instead of z.string. But that doesn't work on the API panel - it shows nothing. Correspondingly, apps/nextjs/src/components/activity/viewactivites.tsx is also modified to fit the format.
2. getSchedule used to be protectedProcedure, but that causes issues with authentication, so temporarily I changed it to publicProcedure, as well as getDailySchedule.

sarinali commented on packages/api/src/router/activity.ts:
```diff
@@ -1,22 +1,52 @@
 export const activityRouter = createTRPCRouter({
-  getSchedule: protectedProcedure
-    .input(z.object({ day: z.date() }))
+  getSchedule: publicProcedure
```
Thanks for fixing this endpoint! I really appreciate it. Good work, last thing to do is switch this back to a `protectedProcedure` since its only for authenticated users.

Sorry! That's the last thing

Adventurer-E commented:
This is really a big thing lol…
And the authentication issue not only happens here, but also in other APIs in the router as well (for example, functions in “announcement” also have such error). I believe this could an issue beyond this ticket.
Could you first approve the merge and later create a new ticket to deal with this issue? I guess this issue should lie in the tRPC panel layer.

From: Sarina Li ***@***.***>
Date: Sunday, November 3, 2024 at 3:08 PM
To: uoftblueprint/centre-for-dreams ***@***.***>
Cc: Eric Lu ***@***.***>, Author ***@***.***>
Subject: Re: [uoftblueprint/centre-for-dreams] eric/ starter+specific-date (PR #485)

@sarinali requested changes on this pull request.

________________________________

In packages/api/src/router/activity.ts<https://github.com/uoftblueprint/centre-for-dreams/pull/485#discussion_r1827056716>:

>

 export const activityRouter = createTRPCRouter({

-  getSchedule: protectedProcedure

-    .input(z.object({ day: z.date() }))

+  getSchedule: publicProcedure

Thanks for fixing this endpoint! I really appreciate it. Good work, last thing to do is switch this back to a protectedProcedure since its only for authenticated users.

Sorry! That's the last thing

—
Reply to this email directly, view it on GitHub<https://github.com/uoftblueprint/centre-for-dreams/pull/485#pullrequestreview-2411934716>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A7CNOZ3PMCFXTUQIVJ4FAWDZ6ZYC5AVCNFSM6AAAAABRC2REH2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDIMJRHEZTINZRGY>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

sarinali commented:
> This is really a big thing lol… And the authentication issue not only happens here, but also in other APIs in the router as well (for example, functions in “announcement” also have such error). I believe this could an issue beyond this ticket. Could you first approve the merge and later create a new ticket to deal with this issue? I guess this issue should lie in the tRPC panel layer. From: Sarina Li ***@***.***> Date: Sunday, November 3, 2024 at 3:08 PM To: uoftblueprint/centre-for-dreams ***@***.***> Cc: Eric Lu ***@***.***>, Author ***@***.***> Subject: Re: [uoftblueprint/centre-for-dreams] eric/ starter+specific-date (PR #485) @sarinali requested changes on this pull request.
> ________________________________ In packages/api/src/router/activity.ts<[#485 (comment)](https://github.com/uoftblueprint/centre-for-dreams/pull/485#discussion_r1827056716)>:
> export const activityRouter = createTRPCRouter({ - getSchedule: protectedProcedure - .input(z.object({ day: z.date() })) + getSchedule: publicProcedure Thanks for fixing this endpoint! I really appreciate it. Good work, last thing to do is switch this back to a protectedProcedure since its only for authenticated users. Sorry! That's the last thing — Reply to this email directly, view it on GitHub<[#485 (review)](https://github.com/uoftblueprint/centre-for-dreams/pull/485#pullrequestreview-2411934716)>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A7CNOZ3PMCFXTUQIVJ4FAWDZ6ZYC5AVCNFSM6AAAAABRC2REH2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDIMJRHEZTINZRGY>. You are receiving this because you authored the thread.Message ID: ***@***.***>

You need to login from localhost:3000

sarinali reviewed (approved):
LGTM
